counsel.el

```diff
@@ -4019,30 +3983,169 @@ TREEP is used to expand internal nodes."
       (goto-char (point-min))
-      (while (re-search-forward outline-regexp nil t)
-        (skip-chars-forward " ")
-        (push (cons (buffer-substring-no-properties
-                     (point) (line-end-position))
-                    (line-beginning-position))
-              cands))
+      (while (re-search-forward (concat "^\\(?:"
+                                        (or (plist-get settings :outline-regexp)
+                                                     outline-regexp)
+                                        "\\)")
+                                nil t)
+        (save-excursion
+          (setq name (or (save-match-data
+                           (funcall (or (plist-get settings :outline-title)
+                                        'counsel-outline-title)))
+                         ""))
+          (goto-char (match-beginning 0))
+          (setq marker (point-marker))
+          (setq level (funcall (or (plist-get settings :outline-level)
+                                   outline-level)))
+          (cond ((eq counsel-org-headline-display-style 'path)
+                     ;; Update stack. The empty entry guards against incorrect
+                     ;; headline hierarchies e.g. a level 3 headline immediately
+                     ;; following a level 1 entry.
+                     (while (<= level stack-level)
+                       (pop stack)
+                       (cl-decf stack-level))
+                     (while (> level stack-level)
+                       (push "" stack)
+                       (cl-incf stack-level))
+                     (setf (car stack) (counsel-org-goto--add-face name level))
+                     (setq name (mapconcat
+                                 #'identity
+                                 (reverse stack)
+                                 counsel-org-headline-path-separator)))
+                    (t
+                     (when (eq counsel-org-headline-display-style 'headline)
+                       (setq name (concat (make-string level ?*) " " name)))
+                     (setq name (counsel-org-goto--add-face name level))))
+          (push (cons name marker) cands))
+        (unless (or (string= name "")
+                    (< orig-point marker))
+          (cl-incf counsel-outline--preselect)))
       (nreverse cands))))
-
+  
 (defun counsel-outline-action (x)
   "Go to outline X."
-  (with-ivy-window
-    (goto-char (cdr x))))
+    (goto-char (cdr x)))
 
 ;;;###autoload
 (defun counsel-outline ()
-  "Jump to outline with completion."
+  "Jump to outline with completion.
+
+This command relies on `outline-regexp', `outline-level', and
+`counsel-outline-title' to match an outline heading and get its
+level and title, respectively.  For major modes where this is not
+adequate or optimal, these variables can be rebound locally in
+the major mode hook.  Replacements are provided for some such
+modes. To set them up, add this to your init file:"
```

Comment:
Should the last sentence be removed, or was it accidentally truncated?

Reply:
Yes the whole paragraph should have been removed, done.